### DISCUSSION TITLE
Undo changes

### DISCUSSION THREAD

**Discussion by lykaodan** [Budibase/budibase#3959] [Help]
How do you "undo" your changes in budibase? Well, as I've been working on my project, and I accidentally delete some important fields , I noticed that I cannot undo my changes. How do you do it? It really takes a lot of time to start all over again. 

Hoping for someone to answer this. Thanks!

**Comment by mjashanks** [comment]
Unfortunately we do not have the feature right now. I feel your pain - it can be very annoying. We are hoping to implement this in the next few months.

**Comment by lykaodan** [comment]
Hello Michael!

I've encountered some problems regarding budibase, can you help me?
Here's the thing, in my project, I have 5 columns, all of them are
integers. I need to sum all of their data per row.
So, I created another column as the sum of all columns.

How should I do it? I tried the {{ add }} formula in budibase, but it only
sums up the first and second column.

I hope you don't mind me asking for your help.

Best regards, have a great day!

**Reply by mjashanks** [reply]
> I hope you don't mind me asking for your help.

Of course not :)  

I think I answered this in [here](https://github.com/Budibase/budibase/discussions/3981) ? Continue the discussion in there, if you have any more questions!

**Comment by lykaodan** [comment]
thank you so much!

**Comment by lykaodan** [comment]
Hi Michael! I'm here again to ask about budibase. I hope you don't mind. :)
So, I'm dealing a project about chickens, and I'm figuring out how to deal
with this formula.

I have a column Mortality (how many chickens died on this day) , Population
(how many chickens added in the pen this day), and Current Population
(overall chickens in the House) .

So, if there are mortalities today and there are no added population, the
current population should be the Last Current population subtracted to the
mortality today.

------------------------------------
Date: 01/13/2022
Current Population: 50

Date: 01/14/2022
Mortality: 10
Population: 5
Current Population: 45

Date: 01/15:2022
Mortality: 5
Population: 0
Current Population: 40
------------------------------------

I hope the illustration above is clear. So, do you have any idea how can I
implement it on Budibase? Thank you so much for all your help!

**Reply by mjashanks** [reply]
Hi @lykaodan 

If you are asking a new question in future, can you please create a new discussion? Thanks :)

However...

I was able to build this. My video below shows the key points. Also, I exported my App so that you could see exactly what I did (attached below the video).

https://user-images.githubusercontent.com/3524181/149549781-e7a4fc4b-f255-4f2b-8436-b677d990514e.mp4

[Chicken coop-export-1642177472255.txt](https://github.com/Budibase/budibase/files/7871745/Chicken.coop-export-1642177472255.txt)
